### ISSUE TITLE
Stacktrace in cp module

### ISSUE THREAD

**Issue by amendlik** [saltstack/salt#52121]
### Description of Issue/Question
Most of the functions in the `cp` modules cause the following stack trace:
```
Exception ignored in: <bound method RemoteClient.__del__ of <salt.fileclient.RemoteClient object at 0x814a45d30>>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/fileclient.py", line 1026, in __del__
  File "/usr/local/lib/python3.6/site-packages/salt/fileclient.py", line 1033, in destroy
  File "/usr/local/lib/python3.6/site-packages/salt/utils/asynchronous.py", line 62, in wrap
  File "/usr/local/lib/python3.6/site-packages/salt/transport/zeromq.py", line 227, in close
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1296, in debug
  File "/usr/local/lib/python3.6/site-packages/salt/log/setup.py", line 343, in _log
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1443, in _log
  File "/usr/local/lib/python3.6/site-packages/salt/log/setup.py", line 363, in makeRecord
NameError: name '__salt_system_encoding__' is not defined
Exception ignored in: <bound method SyncWrapper.__del__ of <salt.utils.asynchronous.SyncWrapper object at 0x814a6cc18>>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/utils/asynchronous.py", line 85, in __del__
  File "/usr/local/lib/python3.6/site-packages/salt/transport/zeromq.py", line 227, in close
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1296, in debug
  File "/usr/local/lib/python3.6/site-packages/salt/log/setup.py", line 343, in _log
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1443, in _log
  File "/usr/local/lib/python3.6/site-packages/salt/log/setup.py", line 363, in makeRecord
NameError: name '__salt_system_encoding__' is not defined
Exception ignored in: <bound method AsyncZeroMQReqChannel.__del__ of <salt.transport.zeromq.AsyncZeroMQReqChannel object at 0x814a59fd0>>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/transport/zeromq.py", line 251, in __del__
  File "/usr/local/lib/python3.6/site-packages/salt/transport/zeromq.py", line 227, in close
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1296, in debug
  File "/usr/local/lib/python3.6/site-packages/salt/log/setup.py", line 343, in _log
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1443, in _log
  File "/usr/local/lib/python3.6/site-packages/salt/log/setup.py", line 363, in makeRecord
NameError: name '__salt_system_encoding__' is not defined
```

### Steps to Reproduce Issue
`salt-call cp.envs`
`salt-call cp.list_master`
`salt-call cp.list_minion`
etc.

### Versions Report
```
Salt Version:
           Salt: 2019.2.0
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: 5.4.0
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.27.8
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.1
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.27.4
         Python: 3.6.8 (default, Feb  8 2019, 21:14:59)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 18.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist:   
         locale: UTF-8
        machine: amd64
        release: 11.2-STABLE
         system: FreeBSD
        version: Not Installed
```

**Comment by amendlik** [comment]
This issue appears to be restricted to the FreeBSD product. I cannot reproduce it on Linux.

**Comment by Ch3LL** [comment]
im not able to replicate this. @s0undt3ch do you have any ideas here?

**Comment by s0undt3ch** [comment]
Might be related to https://github.com/saltstack/salt/pull/52012

**Comment by Ch3LL** [comment]
ahh nice find! thanks @s0undt3ch 

@amendlik can you try that fix?

**Comment by amendlik** [comment]
Applying the patch from 864096b36b seems to have fixed the problem. Can we get that backported to 2019.2?

**Comment by Ch3LL** [comment]
its already been backported to 2018.3 here: https://github.com/saltstack/salt/pull/52035 and will then be merged forward to 2019.2 :)

thanks for confirming the fix :)

**Comment by pkramme** [comment]
Is this already ported to 2019.2?
